src/Workspaces/Remote/ServiceHub/Services/SnapshotService.cs

```diff
@@ -20,9 +20,11 @@ internal partial class SnapshotService : ServiceHubServiceBase
         private AssetSource _source;
 
         public SnapshotService(Stream stream, IServiceProvider serviceProvider) :
-            base(stream, serviceProvider)
+            base(serviceProvider, stream)
         {
             _gate = new object();
+
+            Rpc.StartListening();
```

Comment:
should these all be moved into the base type so that not all subclasses need to remember to do this?